pypdf/_protocols.py

```diff
@@ -67,6 +71,10 @@ class PdfWriterProtocol(Protocol):  # deprecated
     _objects: List[Any]
     _id_translated: Dict[int, Dict[int, int]]
 
+    @property
+    def root_object(self) -> PdfObjectProtocol:
```

Comment:
I propose to move the common protocol elements to a common class as well to reduce duplicates.

Reply:
that will  be for the next steps 😜